### DISCUSSION TITLE
Status of guaranteed delivery for database realtime feature.

### DISCUSSION THREAD

**Discussion by Jonas-Sander** [supabase/supabase#6013] [Questions]
The README of https://github.com/supabase/realtime states

> Does this server guarantee delivery of every data change?
> Not yet! Due to the following limitations:
> 
> Postgres database runs out of disk space due to Write-Ahead Logging (WAL) buildup, which can crash the database and prevent Realtime server from receiving and broadcasting changes. This can be mitigated in the Realtime RLS version of this server by setting the Postgres config max_slot_wal_keep_size to a reasonable size.
> Realtime server can crash due to a larger replication lag than available memory, forcing the creation of a new replication slot and resetting replication to read from the latest WAL data.
> When Realtime server falls too far behind for any reason, for example disconnecting from database as WAL continues to build up, then database can delete WAL segments the server still needs to read from, for example after reconnecting.

Is there some issue where this is tracked? I think it's very important to have guranteed delivery (if thats possible).

Additionally I think this should probably be mentioned in the docs on the website.

**Comment by GaryAustin1** [comment]
It certainly would be nice to know if the connection to the server disconnects in this case.  I assume it does, but don't know for sure.  
You have to deal with connection issues anyways, so as long as it drops connection, the client can know and restart fresh with updated data.
Which the client code needs to handle anyways:  https://github.com/supabase/supabase/discussions/5641

**Comment by w3b6x9** [comment]
@Jonas-Sander

> Is there some issue where this is tracked? I think it's very important to have guranteed delivery (if thats possible).

I just wrote about this here: https://github.com/supabase/supabase/discussions/6697#discussioncomment-3061559

> Additionally I think this should probably be mentioned in the docs on the website.

Good feedback! Will pass on to the rest of the team.
